### ISSUE TITLE
Development environment

### ISSUE THREAD

**Issue by SomeGeek** [Mailu/Mailu#1577]
I'm stuck setting up a development environment for Mailu. I followed the official docs.

https://mailu.io/1.7/contributors/environment.html#web-administration

Everything goes fine until I need to execute run.py which doesn't exist.

So, I tried to run start.py instead, which provides the following error:

`ModuleNotFoundError: No module named 'werkzeug.contrib'`

So, I downgraded werkzeug to 0.16.0. Now I was greeted by the next error:

```
# ./start.py
Usage: flask [OPTIONS] COMMAND [ARGS]...
Try 'flask --help' for help.

Error: No such command 'mailu'.
Usage: flask db upgrade [OPTIONS] [REVISION]

Error: Could not locate a Flask application. You did not provide the "FLASK_APP" environment variable, and a "wsgi.py" or "app.py" module was not found in the current directory.
```

Adding `FLASK_APP=mailu` environment variable gave me the next problem: start.py is stuck. It turns out every flask command just hangs after this variable. No output at all.

```
# ./start.py
<empty line>
*hangs*
```

```
# flask db upgrade
<empty line>
*hangs*
^C
Aborted!
```

(Yes, I'm using Python 3.)

I could be missing something, but the docs seem out of date on this matter. Can someone provide a solution for this?

**Comment by Nebukadneza** [comment]
Hi There,

The `Mailu`-Project is currently in a bit of a bind! We are short on man-power, and we need to judge if it is possible for us to put in some work on this issue.

To help with that, we are currently trying to find out which issues are actively keeping users from using `Mailu`, which issues have someone who want to work on them — and which issues may be less important. These a less important ones could be discarded for the time being, until the project is in a more stable and regular state once again.

In order for us to better assess this, it would be helpful if you could put a **reaction on this post** (use the :smiley: icon to the top-right).
- 👍️ if you **need this** to be able to use Mailu. Ideally, you’d also be able to test this on your installation, and provide feedback …
- 🎉 if you find it a **nice bonus**, but no deal-breaker
- 🚀 if you want to **work on it yourself**!
We want to keep this **voting open for 2 weeks** from now, so please help out!

**Comment by SomeGeek** [comment]
"These a less important ones could be discarded for the time being, until the project is in a more stable and regular state once again."

I mean, you say you are short on man power. Isn't it a bit contra productive that when there _are_ people who actually want to contribute something, they can't even start because of this problem?

**Comment by Diman0** [comment]
I haven't dabbled with the web admin yet. I have tried following the same steps and I also have end result that start.py appears to be hanging.

@kaiyou  Can you provide more info on how to set up a development environment for the admin?
The instructions are outdated.

**Comment by jotjot** [comment]
I have been using Mailu for a while without updating. 
Now, after upgrading my desktop computer, I had to reinstall Mailu and it does not work any more. I did not change anything, configs and all are the same as before. I also do not see any error messages in the normal logs, the webinterface works, but my password is not recognized any more. IMAP access is not possible either. 

Very sad to see this project like this. :-(
